File: easy/3379/main.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* constructTransformedArray(int* nums, int numsSize, int* returnSize) {
  int *res = (int*) malloc(sizeof(int)*numsSize);
  for (int i = 0; i < numsSize; i++) {
      res[i] = nums[((i+nums[i])%numsSize + numsSize)%numsSize];
  }
    
  *returnSize = numsSize;
  return res;
}

File: easy/3379/main.rs
impl Solution {
    pub fn construct_transformed_array(nums: Vec<i32>) -> Vec<i32> {
        let n = nums.len() as i32;
        let mut res = vec![0; n as usize];

        for i in 0..n {
            // usize in vector space
            res[i as usize] = nums[((i + nums[i as usize]).rem_euclid(n)) as usize]
        }
        
        res
    }
}
